dynamiqs/utils/utils.py

```diff
@@ -111,6 +161,52 @@ def unit(x: Tensor) -> Tensor:
     Returns:
         _(..., n, 1) or (..., 1, n) or (..., n, n)_ Normalized ket, bra or density
             matrix.
+
+    Examples:
+        For a ket:
+        >>> psi = dq.fock(4, 0) + dq.fock(4, 1)
+        >>> psi
+        tensor([[1.+0.j],
+                [1.+0.j],
+                [0.+0.j],
+                [0.+0.j]])
+        >>> dq.norm(psi)
+        tensor(1.414)
+        >>> psi = dq.unit(psi)
+        >>> psi
+        tensor([[0.707+0.j],
+                [0.707+0.j],
+                [0.000+0.j],
+                [0.000+0.j]])
+        >>> dq.norm(psi)
+        tensor(1.000)
+
+        For a density matrix:
+        >>> rho = dq.fock_dm(4, 0) + dq.fock_dm(4, 1) + dq.fock_dm(4, 2)
+        >>> rho
+        tensor([[1.+0.j, 0.+0.j, 0.+0.j, 0.+0.j],
+                [0.+0.j, 1.+0.j, 0.+0.j, 0.+0.j],
+                [0.+0.j, 0.+0.j, 1.+0.j, 0.+0.j],
+                [0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j]])
+        >>> dq.norm(rho)
+        tensor(3.)
+        >>> rho = dq.unit(rho)
+        >>> rho
+        tensor([[0.333+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j],
+                [0.000+0.j, 0.333+0.j, 0.000+0.j, 0.000+0.j],
+                [0.000+0.j, 0.000+0.j, 0.333+0.j, 0.000+0.j],
+                [0.000+0.j, 0.000+0.j, 0.000+0.j, 0.000+0.j]])
+        >>> dq.norm(rho)
+        tensor(1.)
+
+        If the argument is batched, each batch element is normalized separately:
+        >>> fock0, fock1 = dq.fock(8, 0), dq.fock(8, 1)
+        >>> x = torch.stack([fock0, fock0 + fock1])  # shape: (2, 8, 1)
+        >>> dq.norm(x)
+        tensor([1.000, 1.414])
+        >>> x = dq.unit(x)                           # shape: (2, 8, 1)
+        >>> dq.norm(x)
+        tensor([1.000, 1.000])
```

Comment:
I think this is overdocumented. It's a simple function, it should have simple documentation.

For instance for the ket part, I think this is already plenty:
```suggestion

    Examples:
        For a ket:
        >>> psi = dq.fock(4, 0) + dq.fock(4, 1)
        >>> dq.norm(psi)
        tensor(1.414)
        >>> psi = dq.unit(psi)
        >>> dq.norm(psi)
        tensor(1.000)
```